coordinator/src/notification_service.rs

```diff
@@ -130,3 +144,205 @@ async fn send_notification<'a>(
     tracing::debug!("Sent: {:?}", response);
     Ok(())
 }
+
+/// Send notifications to users with positions that are about to expire or have
+/// just expired
+pub async fn send_expiry_notifications_if_applicable(
+    positions: &[(Position, FcmToken)],
+    notification_sender: &mpsc::Sender<Notification>,
```

Comment:
:upside_down_face: It would be nicer to test this logic without the `Sender`. You could define a pure function like

```rust
fn expiry_state(position: &Position) -> ExpiryState {
    // ...
}

enum ExpiryState {
    NotExpired,
    SoonToExpire,
    Expired,
}
```
and just test that one.

Reply:
but it's just as easy with the sender? 🤔